test/unit/model/result/ScoreWritterTest.php

```diff
@@ -0,0 +1,120 @@
+                        $var->getCardinality() === taoResultServer_models_classes_OutcomeVariable::CARDINALITY_SINGLE &&
+                        $var->getBaseType() === 'float' &&
+                        $var->getValue() === '0.45';
+                }),
+                'de_id'
+            );
+
+        /** @var ResultServerService|MockObject $resultServiceMock */
+        $resultServiceMock = $this->createMock(ResultServerService::class);
+        $resultServiceMock->expects($this->once())
+            ->method('getResultStorage')
+            ->with('de_id')
+            ->willReturn($resultStorageMock);
+
+        /** @var core_kernel_classes_Resource|MockObject $deliveryMock */
+        $deliveryMock = $this->createMock(core_kernel_classes_Resource::class);
+        $deliveryMock->method('getUri')->willReturn('d_uri');
+
+        /** @var DeliveryExecutionInterface|MockObject $deliveryExecutionMock */
+        $deliveryExecutionMock = $this->createMock(DeliveryExecutionInterface::class);
+        $deliveryExecutionMock->method('getIdentifier')->willReturn('de_id');
+        $deliveryExecutionMock->method('getDelivery')->willReturn($deliveryMock);
+
+        $scoreWritter = new ScoreWriterService();
+        $scoreWritter->setServiceLocator($this->getServiceLocatorMock([
+            ResultServerService::SERVICE_ID => $resultServiceMock
+        ]));
+        $result = $scoreWritter->store($deliveryExecutionMock, '0.45');
+        $this->assertTrue($result);
+    }
+
+    /**
+     * @throws common_exception_Error
+     * @throws common_exception_NotFound
+     */
+    public function testDuplicatedVariable()
+    {
+        /** @var taoResultServer_models_classes_WritableResultStorage|MockObject $resultStorageMock */
+        $resultStorageMock = $this->createMock(taoResultServer_models_classes_WritableResultStorage::class);
+        $resultStorageMock->expects($this->once())
+            ->method('storeTestVariable')
+            ->willThrowException(new DuplicateVariableException('mm'));
+
+        /** @var ResultServerService|MockObject $resultServiceMock */
+        $resultServiceMock = $this->createMock(ResultServerService::class);
+        $resultServiceMock->expects($this->once())
+            ->method('getResultStorage')
+            ->with('de_id')
+            ->willReturn($resultStorageMock);
+
+        /** @var core_kernel_classes_Resource|MockObject $deliveryMock */
+        $deliveryMock = $this->createMock(core_kernel_classes_Resource::class);
+        $deliveryMock->method('getUri')->willReturn('d_uri');
+
+        /** @var DeliveryExecutionInterface|MockObject $deliveryExecutionMock */
+        $deliveryExecutionMock = $this->createMock(DeliveryExecutionInterface::class);
+        $deliveryExecutionMock->method('getIdentifier')->willReturn('de_id');
+        $deliveryExecutionMock->method('getDelivery')->willReturn($deliveryMock);
+
+        $scoreWritter = new ScoreWriterService();
```

Comment:
so the service looks ok but the test name and vars are wrong

Reply:
fixed